app/views/messages/_message.html.erb

```diff
@@ -13,11 +13,31 @@ end %>
 <% scroll_down = scroll_down || initial_page_load %>
 <% thinking ||= last_message && message.content_text == "" && message.not_cancelled? %>
 <% done_thinking = !thinking && message.content_text.present? %>
-<% debug_tool_messages = false %>
+<% memory_updated = from_name_for(message) == "Tool" && message.content_tool_calls.dig(:function, :name) == "memory_remember_detail_about_user" %>
+
+<% if memory_updated %>
+  <div class="flex mb-5 ml-9 sm:ml-3 md:ml-8">
+    <%= icon "archive-box",
+      variant: :outline,
+      size: 22,
+      class: %|
+        ml-1
+        mr-1
+        |
+    %>
+    <div class="ml-1">
+      <%= format_for_display(message.content_text) %>
+    </div>
+    <%= link_to "(manage memories)\n", settings_memories_path,
+      data: { turbo_frame: "_top" },
+      class: "ml-1 text-gray-600 dark:text-gray-300"
+    %>
+  </div>
+<% end %>
```

Comment:
Instead of creating a whole new div structure, let's lean into the existing message structure below. This will help to make future design changes easier and ensure things like "data-role" and "data-subrole" are all in place.

Also, I think it's a bit brittle to do string matching of the tool's response, but I see why you did that because there isn't an easy way to tell what this tool message response is from. That motivated me to just land PR to help you out:
https://github.com/AllYourBot/hostedgpt/pull/563

Here's how I think you can refactor this:

1) Turn "memory_updated?" into a message_helper method so you can use `memory_updated?(message)` wherever you need to within here. I believe it'll simply be this (and I should have done that with some of these extra lines within here):

```
def memory_created?(message) = message.tool? && message.content_tool_calls.dig(:function, :name) == "memory_remember_detail_about_user"
```

2) Update `render_avatar_helper` method to include an elsif for `message.tool?`. Can we make it a wrench-screwdriver? I just played with it locally and this looked nice and visually lined up (although I didn't check dark mode so account for that too):

```
      icon "wrench-screwdriver",
        variant: :micro,
        class: %|
          mt-[5px]
          ml-[2px]
          text-gray-400
        |
```

3) Update `format_for_display` helper to **not** have a "Tool" one since we just want to render this thing as a single line

4) Update the `format_for_display` helper. I think you'll need to pass message into it and then at the very top of it you can parse the JSON and turn it into a link (basically, let's make the whole thing clickable but without an underline and then delete the parenthetical after it). This looked nice in my quick test (again, I didn't look at dark mode):

```
    text = JSON.parse(text) rescue text

    text = link_to(text, settings_memories_path,
      data: { turbo_frame: "_top" },
      class: "text-gray-400 no-underline"
    ) if message
```